core/field.js

```diff
@@ -392,14 +399,39 @@ Blockly.Field.prototype.bindEvents_ = function() {
 };
 
 /**
- * Sets the field's value based on the given XML element. Should only be
- * called by Blockly.Xml.
+ * Disables field if the given XML element specifies this.
  * @param {!Element} fieldElement The element containing info about the
- *    field's state.
+ *     field's state.
+ * @protected
+ */
+Blockly.Field.prototype.setEnabledFromXml_ = function(fieldElement) {
+  if (fieldElement.getAttribute('enabled') === 'false') {
+    this.setEnabled(false);
+  }
+};
+
+/**
+ * Sets the field's value and possibly disables field based on the given XML
+ *     element. Should only be called by Blockly.Xml.
+ * @param {!Element} fieldElement The element containing info about the
+ *     field's state.
  * @package
  */
 Blockly.Field.prototype.fromXml = function(fieldElement) {
   this.setValue(fieldElement.textContent);
+  this.setEnabledFromXml_(fieldElement);
+};
+
+/**
+ * Adds false enabled attribute to the given XML element if field is disabled.
+ * @param {!Element} fieldElement The element containing info about the
+ *     field's state.
+ * @protected
+ */
+Blockly.Field.prototype.addDisabledToXml_ = function(fieldElement) {
```

Comment:
This name is slightly confusing because you're adding an attribute called "enabled". But you're only adding it if the field is disabled so "addEnabledToXml_" might be just as confusing. What do you think?

Reply:
Perhaps `addEnabledFalseToXml_`?  I'm not sure.  After I submitted the pull request, I was also wondering if it'd be better to use something like `addAttributesToXml_` to make it more forward-thinking for potential future additions.  Of course this assumes that the `Field` subclasses would generally add them in the same way.  I'm open to whichever option you think is best.

Reply:
No longer applicable after serialization was reverted.